examples/interactive_local_hyperdrive_example.py

```diff
@@ -1,118 +1,228 @@
-# Here, we execute a trade, where it's calling agent0 + gather data from data pipeline
-# under the hood to allow for error handling and data management
+# Here, we execute a trade on a pool.
 # Return values here mirror the various events emitted from these contract calls
-open_long_event_1 = agent0.open_long(base=FixedPoint(11111))
-open_long_event_1  # pyright: ignore
+# Here, base is unitless and is dependent on the underlying tokens the pool uses.
+open_long_event_1 = agent0.open_long(base=FixedPoint(11111), pool=hyperdrive0)
+
+# %%
+# We can also set an active pool an agent is using to avoid passing in the pool
+# for every trade.
+agent1.set_active(pool=hyperdrive0)
+open_long_event_2 = agent1.open_long(base=FixedPoint(22222))
+
+# Close previous longs from events
+close_long_event_1 = agent0.close_long(
+    maturity_time=open_long_event_1.maturity_time,
+    bonds=open_long_event_1.bond_amount,
+    pool=hyperdrive0,
+)
+
+# Close previous longs from the wallet
+# This wallet is a wallet object that contains positions relevant to a specific pool.
+# This takes a snapshot of the current wallet positions.
+agent0.get_wallet(pool=hyperdrive0)
+# agent1's active pool is set, so no need to pass in a pool.
+agent1.get_wallet()
+# We also provide helper functions to get specific position types.
+agent1_longs = agent1.get_longs()
+close_long_event_2 = agent1.close_long(maturity_time=agent1_longs[0].maturity_time, bonds=agent1_longs[0].balance)
+
+# Shorts
+open_short_event = agent1.open_short(bonds=FixedPoint(33333))
+agent1_shorts = agent1.get_shorts()
+close_short_event = agent1.close_short(maturity_time=agent1_shorts[0].maturity_time, bonds=agent1_shorts[0].balance)
+
+# LP
+add_lp_event = agent1.add_liquidity(base=FixedPoint(44444))
+remove_lp_event = agent1.remove_liquidity(shares=agent1.get_lp())
+
+# The above trades doesn't result in withdraw shares, but the function below allows you
+# to withdrawal shares from the pool.
+# withdraw_shares_event = agent1.redeem_withdraw_share(shares=agent1.get_withdraw_shares())
+
 # %%
 
-# Another long with a different maturity time
-open_long_event_2 = agent0.open_long(FixedPoint(22222))
+# Agents can also execute policies, which encapsulates actions to take on a pool.
+# This requires initializing a policy class from the agent.
+# For example, we set a policy that makes random trades.
+# We can either initialize a policy on initialization (see agent1's initialization)
+# or we can explicitly call `set_policy` to set a policy on an agent.
+# NOTE: `set_policy` overwrites the existing policy.
+# TODO we may be able to set multiple policies on an agent and hot-swap them
+agent0.set_active(
+    policy=PolicyZoo.random,
+    policy_config=PolicyZoo.random.Config(rng_seed=123),
+)
+
+# Execute policy trade on a pool
+# Output event is one of the possible trade events
+agent0_trades = []
+for i in range(10):
+    # NOTE Since a policy can execute multiple trades per action, the output events is a list
```

Comment:
`    # NOTE: Since a policy can execute multiple trades per action, the output events is a list.`